crates/turbopack-node/js/src/transforms/webpack-loaders.ts

```diff
@@ -333,36 +340,111 @@ const transform = (
           emitWarning: makeErrorEmitter("warning", ipc),
           emitError: makeErrorEmitter("error", ipc),
           getLogger(name: unknown) {
-            const logger = (type: unknown, args: unknown) => {
+            const logFn = (logType: string, ...args: unknown[]) => {
               let trace;
-              switch (type) {
+              switch (logType) {
                 case LogType.warn:
                 case LogType.error:
                 case LogType.trace:
-                  trace = cutOffLoaderExecution(new Error("Trace").stack!)
-                    .split("\n")
-                    .slice(3);
+                case LogType.debug:
+                  trace = parseStackTrace(
+                    cutOffLoaderExecution(new Error("Trace").stack!)
+                      .split("\n")
+                      .slice(3)
+                      .join("\n")
+                  );
                   break;
               }
-              const logEntry = {
+
+              ipc.sendInfo({
+                type: "log",
                 time: Date.now(),
-                type,
+                logType,
                 args,
                 trace,
-              };
-
-              this.hooks.log.call(name, logEntry);
+              });
             };
+            let timers: Map<string, [number, number]> | undefined;
+            let timersAggregates: Map<string, [number, number]> | undefined;
 
             // See https://github.com/webpack/webpack/blob/a48c34b34d2d6c44f9b2b221d7baf278d34ac0be/lib/logging/Logger.js#L8
-            // for the full logger interface if this isn't suffecient
-            logger.error = logger.bind(this, LogType.error);
-            logger.warn = logger.bind(this, LogType.warn);
-            logger.info = logger.bind(this, LogType.info);
-            logger.log = logger.bind(this, LogType.log);
-            logger.debug = logger.bind(this, LogType.debug);
-
-            return logger;
+            return {
+              error: logFn.bind(this, LogType.error),
+              warn: logFn.bind(this, LogType.warn),
+              info: logFn.bind(this, LogType.info),
+              log: logFn.bind(this, LogType.log),
+              debug: logFn.bind(this, LogType.debug),
+              asset: (assertion: boolean, ...args: any[]) => {
```

Comment:
```suggestion
              assert: (assertion: boolean, ...args: any[]) => {
```

Should this throw? Otherwise, execution continues past the assertion...